src/quantum/azext_quantum/tests/latest/test_quantum_jobs.py

```diff
@@ -31,6 +31,21 @@ def test_jobs(self):
         targets = self.cmd('az quantum target list -o json').get_output_in_json()
         assert len(targets) > 0
 
+    def test_build(self):
+        result = build(self, target_id='ionq.simulator', project='src\\quantum\\azext_quantum\\tests\\latest\\source_for_build_test\\QuantumRNG.csproj', target_capability='BasicQuantumFunctionality')
+        assert result == {'result': 'ok'}
+
+        self.testfile = open(os.path.join(os.path.dirname(__file__), 'source_for_build_test/obj/qsharp/config/qsc.rsp'))
+        self.testdata = self.testfile.read()
+        self.assertIn('TargetCapability:BasicQuantumFunctionality', self.testdata)
+        self.testfile.close()
+
+        try:
+            build(self, target_id='ionq.simulator', project='src\\quantum\\azext_quantum\\tests\\latest\\source_for_build_test\\QuantumRNG.csproj', target_capability='BogusQuantumFunctionality')
+            assert False
+        except AzureInternalError as e:
+            assert str(e) == "Failed to compile program."
```

Comment:
Nit (meaning it is up to you whether or not to address this comment): If at all possible it would be nice to check that the msbuild message about the target capability being invalid is properly shown to the user.

Reply:
Good suggestion:  I verified this manually, but I will defer expanding the automated test for now and add this to my list of future testing enhancements.  Thanks!